source/features/mute-users.tsx

```diff
@@ -0,0 +1,134 @@
+	select('.unminimized-comment', comment)!.classList.add('d-none');
+}
+
+function unminimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.add('d-none');
+	select('.minimized-comment .Details-element', comment)!.setAttribute('open', 'true');
+	select('.unminimized-comment', comment)!.classList.remove('d-none');
+}
+
+async function onMuteUnmuteClick(event: React.MouseEvent<HTMLButtonElement>): Promise<void> {
+	let mutedUsers = await getMutedUsers();
+
+	const comment = (event.target as HTMLElement).closest('.js-comment-container')!;
+	const viewportOffset = comment.getBoundingClientRect().top;
+	const user = select('.author', comment)!.textContent!;
+
+	if (mutedUsers.includes(user)) {
+		mutedUsers = mutedUsers.filter(mutedUser => mutedUser !== user);
+	} else {
+		mutedUsers.push(user);
+	}
+
+	await setMutedUsers(mutedUsers);
+
+	const avatars = select.all(`.js-discussion .js-comment-container .avatar[alt="@${user}"`);
+	const comments = avatars.map(avatar => avatar.closest('.js-comment-container') as HTMLElement);
+
+	for (const comment of comments) {
+		if (mutedUsers.includes(user)) {
+			minimizeComment(comment);
+		} else {
+			unminimizeComment(comment);
+		}
+	}
+
+	requestAnimationFrame(() => {
+		const newOffset = comment.getBoundingClientRect().top;
+		window.scrollBy(0, newOffset - viewportOffset);
+	});
+}
+
+async function handleMenuOpening(event: DelegateEvent): Promise<void> {
+	const dropdown = select('.show-more-popover', event.delegateTarget.parentElement!)!;
+	const user = select('.author', dropdown.closest('.js-comment-container')!)!.textContent!;
+
+	if (user === currentUser) {
+		return;
+	}
+
+	const mutedUsers = await getMutedUsers();
+	const isMutedUser = mutedUsers.includes(user);
+
+	const existingButton = select('.rgh-mute-unmute-button', event.delegateTarget.parentElement!);
+	if (existingButton) {
+		if (isMutedUser) {
+			existingButton.textContent = 'Unmute user';
+		} else {
+			existingButton.textContent = 'Mute user';
+		}
+
+		return;
+	}
+
+	// Add option to mute or unmute user
+	appendBefore(dropdown, 'a[data-ga-click^="Report"]',
+		<button
+			className="dropdown-item btn-link rgh-mute-unmute-button"
+			role="menuitem"
+			title={`${isMutedUser ? 'Unmute' : 'Mute'} this user`}
+			onClick={onMuteUnmuteClick}>
+			{isMutedUser ? 'Unmute' : 'Mute'} user
+		</button>
+	);
+}
+
+async function minimizeMutedUserComments(): Promise<void> {
+	const mutedUsers = await getMutedUsers();
+
+	for (const comment of select.all('.js-discussion .js-comment-container')) {
+		const user = select('.author', comment)!.textContent!;
+
+		// If the user _is_ muted, minimize their comment
+		if (mutedUsers.includes(user) && user !== currentUser) {
+			minimizeComment(comment);
+		}
+	}
+}
+
+function init(): void {
+	currentUser = getUsername();
```

Comment:
No need to cache this, the function is already called once. Just use `getUsername` 👍